octoprint_mrbeam/printing/comm_acc2.py

```diff
@@ -1640,11 +1641,49 @@ def _gcode_command_for_cmd(self, cmd):
 
         return gcode.group(1)
 
+    def _onExitState(self, current_state, new_state):
+        """
+        Called when the state machine is exiting a state.
+
+        Args:
+            current_state: current state
+            new_state: new state
+
+        Returns:
+            None
+        """
+        # if we're unexpected exiting the printing or paused state to locked show the job cancelled  error notification
+        if (
+            current_state == self.STATE_PRINTING or current_state == self.STATE_PAUSED
+        ) and new_state == self.STATE_LOCKED:
+            self._fire_print_failed()
+            self._show_job_cancelled_due_to_internal_error()
+
+    def _show_job_cancelled_due_to_internal_error(self):
+        """
+        Shows the job cancelled due to internal error notification.
+
+        Returns:
+            None
+        """
+        notification = (
+            _mrbeam_plugin_implementation.user_notification_system.get_notification(
+                notification_id="job_cancelled_due_to_internal_error",
```

Comment:
Maybe it makes sense to have all the notification ids and error codes in a constants file? I don't know if we have more of these...

Reply:
I like the idea, I will create two new files